bin/expressive

```diff
@@ -51,13 +51,23 @@ $commandList = [
     new Module\RegisterCommand('module:register'),
 ];
 
-if (file_exists($cwd . '/config/container.php')) {
-    $commandList[] = new CreateHandler\CreateHandlerCommand('action:create');
-    $commandList[] = new CreateHandler\CreateHandlerCommand('handler:create');
+$version = strstr(Versions::getVersion('zendframework/zend-expressive-tooling'), '@', true);
+//Instace console application
+$application = new Application('expressive', $version);
+
+$containerConfig = $cwd . '/config/container.php';
+if (file_exists($containerConfig)) {
+    $container = require $containerConfig;
+
+    $commandList[] = new CreateHandler\CreateHandlerCommand('action:create', $cwd, $container);
+    $commandList[] = new CreateHandler\CreateHandlerCommand('handler:create', $cwd, $container);
 }
 
-$version = strstr(Versions::getVersion('zendframework/zend-expressive-tooling'), '@', true);
+// Allow users to provide additional commands via a config/commands.php file
+$commandsFile = $cwd.'/config/commands.php';
+if (file_exists($commandsFile)) {
+    (include $commandsFile)($application, $container ?? null);
+}
 
-$application = new Application('expressive', $version);
 $application->addCommands($commandList);
```

Comment:
Move this line up, to below the original conditional that checks for the container file (after line 64). That way users will get an error if they try to add commands that have the same name as built-ins.

Otherwise, looking good! :+1:

Reply:
If this moved to below, line 67 launch a exception because $application now exist, you can see ?

Reply:
I looked through the logic of `Symfony/Components/Console/Application`, and it looks like `addCommands()` delegates to `add()`, which, in turn will grab the command name and alias to set an entry in its internal `$commands`. What this means is that if a command with the same name registers _later_, it will _override_ the functionality of one registered _earlier_.

So, that means if we leave the line here, the built-in commands _cannot_ be overwritten.

The question is then: do we want to provide that ability? This would allow developers to extend our commands or provide replacements specific to their needs, but could cause confusion if a developer is not aware that somebody on their team made those changes.

Reply:
Could we create a new `Application` instance from the `config/commands.php` file so the built-in commands are preserved, and then use some sort of console flag/option to force the script to run the built-in command even if there are custom ones provided? Idk, just a thought!

Reply:
hmm... on my continuous tests lab, i see a wrong on tooling construction... if i need extend or add more commands, the tooling need is a Application based on symfony application, and always needed extend, create other object based on ToolingApplication.... oh my mind...

Reply:
@lpj145 I'm not understanding what you're saying... Is this comment addressing @timothyfisherdev  or something else? Can you give more context, please?